### DISCUSSION TITLE
Request to add support for dynamically switching authentication on or off

### DISCUSSION THREAD

**Discussion by Rmarian** [etcd-io/jetcd#1362] [Ideas]
Hi @lburgazzoli ,

We are using jetcd in our projects.

I made one modification to our fork to enable jetcd to dynamically switch authentication on or off automatically depending on the etcd server configuration without re-creating the client.

This helped us in handling an upgrade scenario easily.

My question is, do you think this change would make sense for the open source client? If yes I could release these changes back to the project (maybe optionally under some flag?)

Thanks,
Radu.

**Comment by lburgazzoli** [comment]
As long as it backward compatible I don't see any issue, please go ahead to send a PR
